misc/freetype/imgui_freetype.cpp

```diff
@@ -790,6 +830,123 @@ void ImGuiFreeType::SetAllocatorFunctions(void* (*alloc_func)(size_t sz, void* u
+} LunasvgPortState, *PLunasvgPortState;
+
+FT_Error ImGuiLunasvgPortInit(FT_Pointer* _state)
+{
+    *_state = new LunasvgPortState();
+    return FT_Err_Ok;
+}
+
+void ImGuiLunasvgPortFree(FT_Pointer* _state)
+{
+    delete *_state;
+}
+
+FT_Error ImGuiLunasvgPortRender(FT_GlyphSlot slot, FT_Pointer* _state)
+{
+    PLunasvgPortState state = *(PLunasvgPortState*)_state;
+
+    // If there was an error while loading the svg in
+    // `ImGuiLunasvgPortPresetSlot`, the renderer hook
+    // still get called, so just returns the error.
+    if (state->err != FT_Err_Ok) return state->err;
+
+    // rows is height, pitch (or stride) equals to width * sizeof(int32)
+    lunasvg::Bitmap bitmap((uint8_t*)slot->bitmap.buffer, slot->bitmap.width, slot->bitmap.rows, slot->bitmap.pitch);
+
+    state->svg->setMatrix(state->svg->matrix().identity()); // Reset the svg matrix to the default value
+    state->svg->render(bitmap, state->matrix);              // state->matrix is already scaled and translated
+
+    state->err = FT_Err_Ok;
+    return state->err;
+}
+
+FT_Error ImGuiLunasvgPortPresetSlot(FT_GlyphSlot slot, FT_Bool cache, FT_Pointer* _state)
+{
+    FT_SVG_Document   document = (FT_SVG_Document)slot->other;
+    PLunasvgPortState state = *(PLunasvgPortState*)_state;
+    FT_Size_Metrics&  metrics = document->metrics;
+
+    // This function is called twice, once in the `FT_Load_Glyph`
+    // and another right before `ImGuiLunasvgPortRender`.
+    // If it's the latter, don't do anything because it's
+    // already done in the former.
+    if (cache) return state->err;
+
+    state->svg = lunasvg::Document::loadFromData((const char*)document->svg_document, document->svg_document_length);
+    if (state->svg == nullptr)
+    {
+        state->err = FT_Err_Invalid_SVG_Document;
+        return state->err;
+    }
+
+    lunasvg::Box box = state->svg->box();
+
+    double scale = std::min(metrics.x_ppem / box.w, metrics.y_ppem / box.h);
+    double xx = (double)document->transform.xx / (1 << 16);
+    double xy = -(double)document->transform.xy / (1 << 16);
+    double yx = -(double)document->transform.yx / (1 << 16);
+    double yy = (double)document->transform.yy / (1 << 16);
+    double x0 = (double)document->delta.x / 64 * box.w / metrics.x_ppem;
+    double y0 = -(double)document->delta.y / 64 * box.h / metrics.y_ppem;
+
+    // This reset the matrix to its default value
+    state->matrix.identity();
+
+    // Scale and transform, we don't translate the svg yet
+    state->matrix.scale(scale, scale);
+    state->matrix.transform(xx, xy, yx, yy, x0, y0);
+    state->svg->setMatrix(state->matrix);
+
+    // Pre-translate the matrix for the rendering step
+    state->matrix.translate(-box.x, -box.y);
+
+    // Get the box again after the transformation
+    box = state->svg->box();
+
+    // Calculate the bitmap size
+    slot->bitmap_left = FT_Int(box.x);
+    slot->bitmap_top = FT_Int(-box.y);
+    slot->bitmap.rows = unsigned int(ceil(box.h));
```

Comment:
here should be (unsigned)(cell(box.h))

Reply:
This is indeed invalid, [Explicit type conversion](https://en.cppreference.com/w/cpp/language/explicit_cast) section 2)
>  The functional-style cast expression consists of a simple type specifier or a typedef specifier (in other words, a single-word type name, that is, cases such as `unsigned int(expression)` and `int*(expression)` are not valid), followed by a comma-separated list of expressions in parentheses.

Interestingly, both MSVC compiler and intellisense did not complain about this, [godbolt](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxBIapAAOqAqETgwe3r56icmOAkEh4SxRMVxxdpgOqUIETMQE6T5%2BXLaY9rkM1bUE%2BWGR0bG2NXUNmc0KQ93BvUX9pQCUtqhexMjsHCYaAILmAMzByN5YANQmO25O48SYrKfYG9tbwQRHLEzBEHP3JgDsVlsbAE4qLRUExnggTjsACJHDQAOiopz%2BDwBTC8RCOqkhMK8DGSwBC6COTwgqzoEAQc0%2BO2RRzpRyuBGWDFhSK%2B3yh9w4C1onAArLw/BwtKRUJw3NZrEcFEsVpgTmYdjxSARNNyFgBrEBmSRwgBsXDM3w0euNUm%2BAL5fP0nEkgrVos4vAUIDiquF3NIcFgSDQLHidGi5EofoD9BiADdkPF4gB9CNcAGxgwETDjWOqPXSLARvCrABqeEwAHcAPLxRicZU0Wip4guiARB0RYK1ACeVd4LeYxDbpYi2gq7uVfrYglLDFoHY9pCwES8wDcYloLu4vCwryM4hn%2BCulQjaYdmFUFXRa2VT1aDtoeAixHbHiwDoIxDwLE7C2BTGACkLJfLlZrjIggiGI7BSMB8hKGoDq6M0BhGCgkqWPot4upACyoPE7SrrwqAHsQr5YOhHwtG0qQuAw7ieI0/hUT0hTFFkSQpAIIxNAkLHtAxfQlGRQ5VBM7F6OUlQCJ0dQ8TMfHjF0wljBMUlMVwCwyssqwSDy/L2jOYocJiWYvAoUZHAmAJwimabPBAuCECQCpKnMvDuloVKkFqkhmHCGh8jsloAgCep8lwAAc3yGnqNocHapBCiKenOq6Kpqgs3qICgqD%2BoGZAUKSmVhv0wBcDsZizpguYFkWZYVkK1Z0HWDZNjO3btp2pAtb2/aDg4bWjowBATlODpzguS60CubUboYwDbiKu4CQeq4isep6pm1l68jON53g%2BGBrCKL5vh%2BfAGD%2Bf7VYB1ayKB4gQfwgiKCo6gzropUIcYyE2NtJGYdhqS4aKBFEYeGH8WJfgQK48mkIEUyMbMnE5Kk0PZKxDBKQjontBJ9Q0aMYPY4pcO8SJQl4xxsmScT0maYs6ngVpHACrFDp6ZiIV6gAtIZwDIMgRzFXCZhHDZ%2BBEMQDkqc5KWaiAOw7HC8tK8rKtRTFcV4U6thJS56pRWYOnxVrutuQRyTOJIQA).

![image](https://github.com/ocornut/imgui/assets/136492105/1cafd8b5-3fcb-4a0c-a838-32d12da60045)